tests/console/snapper_cleanup_timeline.pm

```diff
@@ -36,9 +36,13 @@ sub convert2numeric {
     return $str2convert;
 }
 
-sub get_used_disk_space {
-    my $used_space = script_output("df -h /  | awk \'NR==2 {print \$5}\'");
-    return convert2numeric($used_space);
+# returns the total disk size number without units, if the given argument is "size".
+# Othern wise it returns the percentage % of used disk space.
+sub get_disk {
+    my ($args) = @_;
+    my $column     = $args eq "size" ? 2 : 5;
```

Comment:
I was referring to have a hash, because 'if/else' is not extensible.

Reply:
we just need one argument here, why would we want it to be extensible?

Reply:
That was the idea of the suggestion, so we could reuse in the future in some library as 'df' is used a lot.
So if we have for instance:
``'
df -h /                                 
Filesystem      Size  Used Avail Use% Mounted on
/dev/sda2       244G  124G  119G  52% /
```
we could implement something like this to map columns default columns: 'Size', 'Used', 'Avail', 'Use'.
```
sub get_disk_size {
   my $col_name= shift;
   my %col_index = (
        size  => 2,
        used => 3,
        avail => 4,
        use   => 5
    );
    my $index = $col_index{col_name};
    die 'column name not mapped for dh command' unless $index;
    my $size = script_output("df -h /  | awk \'NR==2 {print \$$index}\'");
    return convert2numeric($size);
}
```
Although after writing this I think best solution would be to use an array instead of a hash to have a list with all default columns in the exact order and then get the index from that list.
```
use List::MoreUtils qw(first_index);
...
sub get_disk_space_usage {
...
my @field_list = qw(
   file_system
   size,
   used,
   avail,
   use,
   mounted_on
);
my $index = firstidx { $_ eq $col_name } @field_list;
die 'no such item or more than one of these has been found.' if $index == -1;
my $size = script_output("df -h /  | awk \'NR==2 {print \$$index}\'");
```
Then we could get any field from the output of `df` and 'convert2numeric' is applied outside the function when needed.

Another advantage is that for the future if we someone needs extend this to use the command with different set of FIELD_LIST (see man page) we can pass the list in the exact order as argument.

Reply:
In case you want to save this suggestion for other PR, not this one, it is already good we don't have now two methods, so it is an improvement, at least I would recommend to rename current method to `get_disk_usage` due we are not getting the disk, we are getting the usage of the disk in two formats, might be more readable.

Reply:
I created separate lib/disk_utils.pm where I combine your proposal with Rodion's